snapcraft/commands/validation_sets.py

```diff
@@ -174,25 +190,33 @@ def _generate_template(
     return unverified_validation_sets
 
 
-def edit_validation_sets(validation_sets_path: Path) -> Dict[str, Any]:
+def edit_validation_sets(
+    validation_sets_path: Path,
+) -> validation_sets.EditableBuildAssertion:
     """Spawn an editor to modify the validation-sets."""
     editor_cmd = os.getenv("EDITOR", "vi")
 
     while True:
         with emit.pause():
             subprocess.run([editor_cmd, validation_sets_path], check=True)
         try:
-            edited_validation_sets = yaml.safe_load(
-                validation_sets_path.read_text(encoding="utf-8")
+            with validation_sets_path.open() as file:
+                data = craft_application.util.safe_yaml_load(file)
+            edited_validation_sets = validation_sets.EditableBuildAssertion.from_yaml_data(
+                data=data,
+                # filepath is only shown for pydantic errors and snapcraft should
+                # not expose the temp file name
+                filepath=Path("validation-sets"),
             )
             return edited_validation_sets
-        except yaml.YAMLError as yaml_error:
-            emit.message(f"A YAML parsing error occurred {yaml_error!s}")
+        except (yaml.YAMLError, CraftValidationError) as err:
+            emit.message(f"{err!s}")
             if not utils.confirm_with_user("Do you wish to amend the validation set?"):
-                raise errors.SnapcraftError("Operation aborted") from yaml_error
+                raise errors.SnapcraftError("Operation aborted") from err
```

Comment:
We have established (during the rockcraft error message revamp) that error messages should start with lowercase characters for consistency.

Reply:
Fixed the existing errors.